packages/block-library/src/navigation/edit/index.js

```diff
@@ -226,16 +227,18 @@ function Navigation( {
 
 	const { getNavigationFallbackId } = unlock( useSelect( coreStore ) );
 
-	const navigationFallbackId = ! ( ref || hasUnsavedBlocks )
-		? getNavigationFallbackId()
-		: null;
-
 	useEffect( () => {
 		// If:
+		// - we haven't resolved the menus request and there isn't one missing, OR
 		// - there is an existing menu, OR
 		// - there are existing (uncontrolled) inner blocks
 		// ...then don't request a fallback menu.
-		if ( ref || hasUnsavedBlocks || ! navigationFallbackId ) {
+		if (
+			! navMenuResolvedButMissing ||
+			navigationMenus.length > 0 ||
```

Comment:
"Fallback" is an interesting concept. It doesn't mean "create a fallback". I mean "get the most appropriate menu to use as a fallback".

This will run

- if there are no menus at all
- if there are menus but this particular nav block does not have a `ref`

Prior to this we just had lots of "use the first menu" scattered all over the place so the fallback `selector` helps to unify this.

That said, if the selector doesn't immediately return data from state, then it will trigger a resolver which is async and thus we do need to conditionalise it.

Reply:
Hmmm... So we need to make a request for the fallback if there are no menus, or **if we don't know what menu they want.** So, it was decided to be better UX to show _some menu_ over an error/no menu?

Reply:
Yes that's it. So we want to avoid a "no menus" situation. 

So we 

- get the most recently created Navigation Menu post (if available). If not we fall back to...
- convert a most recently created Classic Menu to a Navigation Menu and use that. If no classic menu we fall back to...
- create a Navigation Menu and use that.

That logic is all server side in a dedicated class that is called from the fallback endpoint.